sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuRangePartitioner.scala

```diff
@@ -195,32 +198,37 @@ case class GpuRangePartitioner(
     }
   }
 
-  def computeBoundsAndClose(batch: ColumnarBatch): (Array[Int], Array[GpuColumnVector]) = {
+  def computeBoundsAndCloseWithRetry(batch: ColumnarBatch): (Array[Int], Array[GpuColumnVector]) = {
     val types = GpuColumnVector.extractTypes(batch)
-    val partedTable = withResource(batch) { batch =>
-      val parts = withResource(new NvtxRange("Calculate part", NvtxColor.CYAN)) { _ =>
-        computePartitionIndexes(batch)
-      }
-      withResource(parts) { parts =>
-        withResource(GpuColumnVector.from(batch)) { table =>
-          table.partition(parts, numPartitions)
+    val spillableBatch = SpillableColumnarBatch(GpuColumnVector.incRefCounts(batch),
```

Comment:
Doesn't this leak?  It looks like we're explicitly trying to avoid batch from being exclusively owned by the SpillableColumnarBatch instance but then never reference `batch` again.  This method is responsible for closing `batch` (as implied by the method name) yet I don't see where that happens.